src/client/hooks/useKeyEvent/index.tsx

```diff
@@ -7,19 +7,24 @@ import { blockTypes } from '/client/constants/tetriminos';
 import { keyboard } from '/client/constants/keyboard';
 import Block from '/client/models/block';
 
-export const useKeyEvent = ({ setIsPause, setMatrix, setBlock }
+export const useKeyEvent = ({ setIsPause, setMatrix, setBlock, setCurrentPlayer }
   : {
     setIsPause: Dispatch<SetStateAction<boolean>>,
     setMatrix: Dispatch<SetStateAction<Matrix>>,
     setBlock: Dispatch<SetStateAction<Block>>,
+    setCurrentPlayer: Dispatch<SetStateAction<PlayerType | null>>,
   }): void => {
   useKey((_key: number, { keyCode }: { keyCode: number }) => {
-    if (keyCode === keyboard.pause) setIsPause((prevState) => !prevState);
-    if (keyCode === keyboard.reset) {
-      setBlock(new Block({ type: _.sample(blockTypes) ?? 'T' }));
-      setMatrix(blankMatrix);
-      setIsPause(true);
-    }
+    setCurrentPlayer((currentPlayer?) => { 
+      console.log('Leader 2:', currentPlayer?.isLeader); // TODO: prop is not visible here, how to pass it?
+      if (keyCode === keyboard.pause && currentPlayer?.isLeader) setIsPause((prevState) => !prevState);
```

Comment:
After isLeader started working I can only toggle setIsPause with P key twice, after that it stops working
https://www.loom.com/share/c7a948586a9441f7a7464d2d57e4e0b2